libres/lib/include/ert/enkf/block_fs_driver.hpp

```diff
@@ -18,27 +18,57 @@
 
 #ifndef ERT_BLOCK_FS_DRIVER_H
 #define ERT_BLOCK_FS_DRIVER_H
-#ifdef __cplusplus
-extern "C" {
-#endif
 
 #include <stdio.h>
 #include <stdbool.h>
 
 #include <ert/enkf/fs_types.hpp>
 
-typedef struct block_fs_driver_struct block_fs_driver_type;
+typedef struct buffer_struct buffer_type;
+typedef struct bfs_config_struct bfs_config_type;
+typedef struct bfs_struct bfs_type;
+
+namespace ert {
+
+class block_fs_driver {
+    int num_fs;
+    bfs_config_type *config{};
+    bfs_type **fs_list;
+
+public:
+    block_fs_driver(int num_fs);
+    ~block_fs_driver();
+
+    static block_fs_driver *new_(bool preload, bool read_only, int num_fs,
+                                 const char *mountfile_fmt,
+                                 bool block_level_lock);
+    static block_fs_driver *open(FILE *fstab_stream, const char *mount_point,
+                                 bool preload, bool read_only);
+
+    bool has_node(const char *node_key, int report_step, int iens);
```

Comment:
Consider making these `const`.

Reply:
This would require making `get_fs` const, which makes no sense. I plan to ground-up rewrite blockfs soon so I don't think it's necessary at this point.